nixops/resources/sns_topic.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+
+# Automatic provisioning of AWS SNS topics.
+
+import boto
+import boto.sns
+import nixops.util
+import nixops.resources
+import nixops.ec2_utils
+from xml.etree import ElementTree
+
+class SNSTopicDefinition(nixops.resources.ResourceDefinition):
+    """Definition of an SNS topic."""
+
+    @classmethod
+    def get_type(cls):
+        return "sns-topic"
+
+    @classmethod
+    def get_resource_type(cls):
+        return "snsTopics"
+
+    def __init__(self, xml):
+        nixops.resources.ResourceDefinition.__init__(self, xml)
+        self.topic_name = xml.find("attrs/attr[@name='name']/string").get("value")
```

Comment:
@edolstra I removed the xml stuff and tested again. I also added one missing edge case handling which is when the sns topic is removed from the console so that it gets created again.